### DISCUSSION TITLE
Eigenmode coefficient increases along the length of a straight waveguide? How does get_eigenmode_coefficients

### DISCUSSION THREAD

**Discussion by peterropac** [NanoComp/meep#2762] [Q&A] [answered]
I have an eigenmode Gaussian source at the start of a 2D straight waveguide emitting the first eigenmode. After that I have two mode monitors created with `sim.add_mode_monitor` at different distances after the source, oriented perpendicular to the direction of the waveguide across the width of the waveguide.
![Figure_1](https://github.com/NanoComp/meep/assets/112213064/59a55f61-2744-4fca-808c-56e3b6897874)

I get the eigenmode coefficients of the first eigenmode at both *mode monitors*, and when I calculate the ratio between the two coefficient I notice that at certain wavelenghts the eigenmode coefficient of the furthermost *mode monitor* is greater than the eigenmode coefficient that i get from the closer *mode monitor*. How is this possible? How does `sim.get_eigenmode_coefficients` calculate coeffecients?

Numerical example:
The center wavelength of the source is 1.55um and `fwidth = 0.2`. I measure the eigenmode coeffecients at frequncies **0.58064516 (1.722 um), 0.64516129 (1.55 um), 0.70967742 (1.4091um)**, and the ratios `right_eigenmode_coeff/left_eigenmode_coeffs` are **0.80500448, 0.65725958, 1.08718659**.

**Comment by stevengj** [comment]
The ratios are complex numbers.  I assume you are reporting to the absolute values (or absolute values squared)?  Have you checked that the results are converged with resolution (i.e. doubling the resolution doesn't change anything)?

**Reply by peterropac** [reply]
Sorry, actually the ratios are `abs(right_eigenmode_coeff)²/abs(left_eigenmode_coeff)²`
I have doubled the resolution (120 points/um) but nothing changed. I did notice that the values change (except for the one at 1.55um) with respect to the `fwidth` parameter.

**Comment by peterropac** [comment, marked as answer]
In case anyone faces the same problem:

- It helps to broaden the source. In my case `fwidth = 0.3` was good enough.
- it helps to run the simulation with `mp.stop_when_fields_decayed` and to set `decay_by` to something like 10^-8. I had it at 10^-4 and not all fields sufficiently decayed. 10^-8 is way less than I thought I would need...

Sorry for cluttering the discussions.
